source/frontend/src/components/Navbar.jsx

```diff
@@ -5,49 +5,63 @@ import PropTypes from 'prop-types'
 import Login from './login';
 import Logout from './logout';
 import { useGoogleDocs } from '../GoogleDocsContext';
+import placeholderImageUrl from './placeholder.png'; // Adjust the path if necessary
 
 
 
 
 function Navbar() {
     const { comments, accessToken } = useGoogleDocs();
     const [isUserLoggedIn, setIsUserLoggedIn] = useState(false);  // State to track login status
+    const [profileImageUrl, setProfileImageUrl] = useState(null);
+    const [userName, setUsername] = useState(null);
+    // const placeholderImageUrl = './placeholder.png';  // Placeholder image URL
+
 
 
     const onSuccessLogin = (res) => {
         console.log("LOGIN SUCCESS! Current user :", res.profileObj);
-        console.log("badaboom");
         setIsUserLoggedIn(true);  // Set login status to true on successful login
-
+        console.log(res.profileObj.imageUrl);
+        setProfileImageUrl(res.profileObj.imageUrl);
+        setUsername(res.profileObj.name);
     };
 
     const onSuccessLogout = (res) => {
```

Comment:
The onSuccesLogout  lets us know when the user has logged out but it does not clear out the 'profileimageurl' and 'username' state which could eventually be leaked.
'Profileimageurl' and 'username' should be reverted back to original state
Below code could be added
const onSuccessLogout = (res) => {
    console.log("Logout successful");
    setIsUserLoggedIn(false);  
    setProfileImageUrl(null); 
    setUsername(null);         
};